app/models/income_model.rb

```diff
@@ -36,9 +36,13 @@ def relevant_frequency_options
   end
 
   def income_frequency_options
+    IncomeModel.income_frequency_options(skip_year: controlled)
+  end
+
+  def self.income_frequency_options(skip_year:)
     standard = STANDARD_FREQUENCY_OPTIONS.map { [_1, I18n.t("estimate_flow.income.frequencies.#{_1}")] }
     addendum = [:divider, [THREE_MONTHS, I18n.t("estimate_flow.income.frequencies.#{THREE_MONTHS}")]]
-    return standard + addendum if controlled
+    return standard + addendum if skip_year
```

Comment:
Ah, ok, so we have an _instance_ method and a _class_ method which have the same name. The instance method calls the class method. The instance method knows whether this instance is controlled or certificated. The class method only cares about whether or not to skip the "Total in the last year" option. This way, our new DependantIncomeModel can _also_ call the IncomeModel class method, so as to avoid duplicating code. (But there seems to be a little bug in the code that you've spotted in your headline review comment, which I need to fix).

Reply:
I'll be thinking this one over, but now get the gist of it. Thank you.